spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -0,0 +1,353 @@
+          } else {
+            return greaterThan(
+                unquote(predicate.children()[1].toString()),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[0]).value()));
+          }
+
+        case LT_EQ:
+          if (predicate.children()[1] instanceof LiteralValue) {
+            return lessThanOrEqual(
+                unquote(predicate.children()[0].toString()),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[1]).value()));
+          } else {
+            return greaterThanOrEqual(
+                unquote(predicate.children()[1].toString()),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[0]).value()));
+          }
+
+        case GT:
+          if (predicate.children()[1] instanceof LiteralValue) {
+            return greaterThan(
+                unquote(predicate.children()[0].toString()),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[1]).value()));
+          } else {
+            return lessThan(
+                unquote(predicate.children()[1].toString()),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[0]).value()));
+          }
+
+        case GT_EQ:
+          if (predicate.children()[1] instanceof LiteralValue) {
+            return greaterThanOrEqual(
+                unquote(predicate.children()[0].toString()),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[1]).value()));
+          } else {
+            return lessThanOrEqual(
+                unquote(predicate.children()[1].toString()),
+                convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[0]).value()));
+          }
+
+        case EQ: // used for both eq and null-safe-eq
+          Object value;
+          String attributeName;
+          if (predicate.children()[1] instanceof LiteralValue) {
+            attributeName = predicate.children()[0].toString();
+            value = convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[1]).value());
+          } else {
+            attributeName = predicate.children()[1].toString();
+            value = convertUTF8StringIfNecessary(((LiteralValue) predicate.children()[0]).value());
+          }
+
+          if (predicate.name().equals("=")) {
+            // comparison with null in normal equality is always null. this is probably a mistake.
+            Preconditions.checkNotNull(
+                value, "Expression is always false (eq is not null-safe): %s", predicate);
+            return handleEqual(unquote(attributeName), value);
+          } else if (predicate.name().equals("<=>")) {
+            if (value == null) {
+              return isNull(unquote(attributeName));
+            } else {
+              return handleEqual(unquote(attributeName), value);
+            }
+          }
+          break;
```

Comment:
Nit: maybe we should return null explicitly like the other case instead of break.  Also curious, will this ever happen?

Reply:
This will not happen. I removed this.